lib/ace/transport_app.rb

```diff
@@ -16,6 +19,7 @@ def initialize(config = nil)
       @config = config
       @executor = Bolt::Executor.new(0, load_config: false)
       @file_cache = BoltServer::FileCache.new(@config).setup
+      @plugins = ACE::PluginCache.new(@config).setup
```

Comment:
Please revert the `environments-dir` change, and instead pass

```
@configs.merge('cache-dir' => File.join(@configs['cache-dir'], 'tasks')
```

and 

```
@configs.merge('cache-dir' => File.join(@configs['cache-dir'], 'environments')
```

respectively. This will also create new `Hash` instances avoiding the two utility classes to interfere with each others config.